src/views/PrivateRoutes.jsx

```diff
@@ -0,0 +1,28 @@
+import { useState, useEffect } from "react";
+import { Navigate, Outlet } from "react-router-dom";
+import { onAuthStateChanged } from "firebase/auth";
+import { auth } from "../../firebase_config";
+
+function PrivateRoutes() {
+  const [user, setUser] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    onAuthStateChanged(auth, (user) => {
+      setIsLoading(false);
+      if (user) {
+        setUser(user);
+      } else {
+        setUser(null);
+      }
+    });
+  }, []);
+
+  if (isLoading) {
```

Comment:
we may want to consider adding some styling or a spinner for the loading state to enhance the user experience.
I found this cool npm package for different spinners: [npm page for react-spinners](https://www.npmjs.com/package/react-spinners) or [demo page for react-spinners](https://www.davidhu.io/react-spinners/)